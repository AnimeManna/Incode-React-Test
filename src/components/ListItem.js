import React, {Component} from 'react'

import {
    List,
    Segment
} from 'semantic-ui-react'

class ListItem extends Component {
    render() {
        console.log(this.props)
        return (
            <List.Item>
                <List.Icon name={this.props.icon}/>
                <List.Content> {this.props.firstText} {this.props.secondText} {this.props.thirdText} {this.props.fourthText}</List.Content>
            </List.Item>
        )
    }
}

export default ListItem