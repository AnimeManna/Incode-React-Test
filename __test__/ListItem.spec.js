import React from 'react'
import configureStore from 'redux-mock-store'
import renderer from 'react-test-renderer'
import {shallow, render, mount} from 'enzyme'

import ListItem from '../src/components/ListItem'
import toJson from "enzyme-to-json";

describe('ListItem container', () => {
    const props = {
        user: {
            general: {
                firstName: 'Ololoev',
                lastName: 'Ololevn',
                avatar: 'https://s3.amazonaws.com/uifaces/faces/twitter/andysolomon/128.jpg'
            },
            job: {
                title: 'DNU',
                company: 'Incode-Group'
            },
            contact: {
                email: 'ololoev@gmail.com',
                phone: '0999201'
            },
            address: {
                country: 'Ukraine',
                city: 'Dnepr',
                street: 'Kazakova',
                zipCode: '49000'

            }
        }
    }
    const {general} = props.user
    describe('render()',()=>{

        test('render the component',()=>{
            const wrapper = shallow(<ListItem icon="user" firstText={general.firstName} secondText={general.lastName}/>)
            const container = wrapper.dive()

            expect(toJson(container)).toMatchSnapshot();
        })
    })

})

