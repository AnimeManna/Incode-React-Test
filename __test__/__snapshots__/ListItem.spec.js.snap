// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Client container render() render the component 1`] = `
<div
  className="item"
  onClick={[Function]}
  role="listitem"
>
  <ListIcon
    name="user"
  />
  <ListContent>
     
    Ololoev
     
    Ololevn
     
     
  </ListContent>
</div>
`;

exports[`ListItem container render() render the component 1`] = `
<div
  className="item"
  onClick={[Function]}
  role="listitem"
>
  <ListIcon
    name="user"
  />
  <ListContent>
     
    Ololoev
     
    Ololevn
     
     
  </ListContent>
</div>
`;
